unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/AssetBundles/AB/AssetPromise_AB.cs

```diff
@@ -24,35 +22,17 @@ public class AssetPromise_AB : AssetPromise_WithUrl<Asset_AB>
         public static int downloadingCount => concurrentRequests;
         public static int queueCount => AssetPromiseKeeper_AB.i.waitingPromisesCount;
 
-        static readonly float maxLoadBudgetTime = 0.032f;
-        static float currentLoadBudgetTime = 0;
-
-        public static bool limitTimeBudget => CommonScriptableObjects.rendererState.Get();
-
         Coroutine loadCoroutine;
         static HashSet<string> failedRequestUrls = new HashSet<string>();
 
         List<AssetPromise_AB> dependencyPromises = new List<AssetPromise_AB>();
         UnityWebRequest assetBundleRequest = null;
 
-        static Dictionary<string, int> loadOrderByExtension = new Dictionary<string, int>()
-        {
-            {"png", 0},
-            {"jpg", 1},
-            {"peg", 2},
-            {"bmp", 3},
-            {"psd", 4},
-            {"iff", 5},
-            {"mat", 6},
-            {"nim", 7},
-            {"ltf", 8},
-            {"glb", 9}
-        };
-
-        private IOrderedEnumerable<string> assetsToLoad;
+        public static AssetBundlesLoader assetBundlesLoader = new AssetBundlesLoader();
 
         public AssetPromise_AB(string contentUrl, string hash) : base(contentUrl, hash)
         {
+            assetBundlesLoader.Start();
```

Comment:
Beware! here you are re-creating the loader for every promise that gets loaded! This needs a null check.

Reply:
I know, that's why I added that null check on the loader `Start()` function:

```
public void Start()
{
     if (assetBundlesLoadingCoroutine != null)
          return;

     assetBundlesLoadingCoroutine = CoroutineStarter.Start(LoadAssetBundlesCoroutine());
}
```